File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/PayoutCompletedCommand.java
package com.hedvig.paymentservice.domain.payments.commands;

import java.time.Instant;
import java.util.UUID;

import javax.money.MonetaryAmount;

import org.axonframework.commandhandling.TargetAggregateIdentifier;

import lombok.Value;

@Value
public class PayoutCompletedCommand {
    @TargetAggregateIdentifier
    String memberId;

    UUID transactionId;
    MonetaryAmount amount;
    Instant timestamp;
}

File: payment-service/src/main/java/com/hedvig/paymentservice/domain/payments/commands/PayoutCompletedCommand.kt
package com.hedvig.paymentservice.domain.payments.commands

import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.time.Instant
import java.util.UUID
import javax.money.MonetaryAmount

data class PayoutCompletedCommand(
  @TargetAggregateIdentifier
  val memberId: String,
  val transactionId: UUID,
  val amount: MonetaryAmount,
  val timestamp: Instant
)
